u19_pipeline/recording.py

```diff
@@ -0,0 +1,181 @@
+import datajoint as dj
+from u19_pipeline import lab, task, subject, acquisition
+
+schema = dj.schema('u19_recording')
+
+
+
+@schema
+class RecordingModality(dj.Lookup):
+     definition = """
+     recording_modality:        varchar(64)          # modalities for recording (ephys, imaging, video_recording, etc.) 
+     ---
+     modality_description:      varchar(255)         # description for the modality
+     root_direcory:             varchar(255)         # root directory where that modality is stored (e.g. ephys = /braininit/Data/eletrophysiology)
+     recording_file_extensions: blob                 # file extensions specific for this modality
+     """
+     contents = [
+        ['ephys',             '', '/braininit/Data/electrophysiology', ['ap.bin', 'ap.meta']],
+        ['imaging',           '', '/braininit/Data/imaging', ['.avi', '.tiff','.tif']],
+        ['video_acquisition', '', '/braininit/Data/video_acquisition', ['.avi', '.mp4']]
+     ]
+
+
+@schema
+class PreprocessingParamSet(dj.Lookup):
+    definition = """
+    # Parameter set to be used in the preprocessing steps
+    preprocess_paramset_idx:  smallint
+    ---
+    -> RecordingModality    
+    preprocess_paramset_desc: varchar(128)
+    preprocess_param_set_hash: uuid
+    unique index (preprocess_param_set_hash)
+    preprocessing_params: longblob  # dictionary of all applicable parameters
+    """
+
+    @classmethod
+    def insert_new_params(cls, recording_modality: str, paramset_idx: int,
+                          paramset_desc: str, params: dict):
+        param_dict = {'recording_modality': recording_modality,
+                      'preprocess_paramset_idx': paramset_idx,
+                      'preprocess_paramset_desc': paramset_desc,
+                      'preprocessing_params': params,
+                      'preprocess_param_set_hash':  dict_to_uuid(params)}
+        param_query = cls & {'preprocess_param_set_hash': param_dict['preprocess_param_set_hash']}
+
+        if param_query:  # If the specified param-set already exists
+            existing_paramset_idx = param_query.fetch1('preprocess_paramset_idx')
+            if existing_paramset_idx == paramset_idx:  # If the existing set has the same paramset_idx: job done
```

Comment:
```suggestion
    @classmethod
    def insert_new_params(cls, recording_modality: str, preprocess_paramset_idx: int,
                          preprocess_paramset_desc: str, preprocess_paramset: dict):
        paramset_dict = {'recording_modality': recording_modality,
                      'preprocess_paramset_idx': preprocess_paramset_idx,
                      'preprocess_paramset_desc': preprocess_paramset_desc,
                      'preprocess_paramset': preprocess_paramset,
                      'preprocess_paramset_hash':  dict_to_uuid(preprocess_paramset)}
        paramset_query = cls & {'preprocess_paramset_hash': param_dict['preprocess_paramset_hash']}

        if paramset_query:  # If the specified param-set already exists
            existing_paramset_idx = paramset_query.fetch1('preprocess_paramset_idx')
            if existing_paramset_idx == preprocess_paramset_idx:  # If the existing set has the same preprocess_paramset_idx: job done
```

Reply:
I have changed this to match the proposed new names.